app/presenters/confirmation_letter_presenter.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+class ConfirmationLetterPresenter
+  delegate :reference, to: :registration
+
+  def initialize(registration)
+    @registration = registration
+  end
+
+  def sorted_active_registration_exemptions
+    registration_exemptions_with_exemptions.where(state: :active).order(:exemption_id)
+  end
+
+  def sorted_deregistered_registration_exemptions
+    registration_exemptions_with_exemptions.where("state != ?", :active).order_by_state_then_exemption_id
+  end
+
+  # Used only if the page contains a single letter.
+  def webpage_title
+    I18n.t("confirmation_letter.single_title", reg_number: registration.reference)
+  end
+
+  def date_of_letter
+    I18n.l(Time.zone.today, format: :long)
+  end
+
+  # Provides the full postal address for the letter.
+  def postal_address_lines
+    [
+      "#{registration.contact_first_name} #{registration.contact_last_name}",
+      registration.operator_name,
+      address_lines(registration.contact_address)
+    ].flatten!.reject(&:blank?)
+  end
+
+  def reg_details_items
+    submission_date = I18n.l(registration.submitted_at.to_date, format: :long)
+    filter_blank_items([
+                         { key: t("reg_details_reference"), value: registration.reference },
```

Comment:
This causes a translation error. I think this would fix it:
```suggestion
                         { key: t("reg_details_reference_number"), value: registration.reference },
```